drivers/flash/soc_flash_mcux.c

```diff
@@ -216,7 +223,128 @@ static int flash_mcux_write(const struct device *dev, off_t offset,
 	addr = offset + priv->pflash_block_base;
 
 	key = irq_lock();
+#ifdef CONFIG_FLASH_MCUX_LPC_IAP_WRITE_BUFFER
+	size_t page_size = DT_PROP(SOC_NV_FLASH_NODE, write_block_size);
+	size_t size_before = offset % page_size;
```

Comment:
Would be good to have comment that it refers to page data before the offset on which incoming record begins.
Will help to read this algorithm for anyone in the feature.
I see similar note to other helper variables